### PR TITLE
Documentation for adding configuration to a lint and common abbreviations

### PR DESCRIPTION
**Pull request by xFrednet**
This PR adds some commonly used abbreviations to the `basis.md` file and a guide on how to implement a configuration value for a lint. 

* [Rendered `/doc/basics.md` (Abbreviation list)](https://github.com/xFrednet/rust-clippy/blob/0000-configuration-documentation/doc/basics.md#common-abbreviations)
* [Rendered `/doc/adding_lints.md` (Configuration value guide)](https://github.com/xFrednet/rust-clippy/blob/0000-configuration-documentation/doc/adding_lints.md#adding-configuration-to-a-lint)

I'm not sure if the guide is written in the best way. Style suggestions are appreciated. :upside_down_face: 

 ---

Again a big **thank you** for everyone who helped to collect the abbreviation list over on [zulip]. I had a lot of fun, and it was also very informative. Keep up the good work :upside_down_face: 

[zulip]: https://rust-lang.zulipchat.com/#narrow/stream/257328-clippy/topic/Common.20abbreviations.20in.20basics.2Emd/near/223548065
---

changelog: none

### CODE REVIEW

File: doc/adding_lints.md [hunk 1]
```diff
@@ -526,6 +527,81 @@ Before submitting your PR make sure you followed all of the basic requirements:
...
+
+Clippy supports the configuration of lints values using a `clippy.toml` file in the workspace 
+directory. Adding a configuration to a lint can be useful for thresholds or to constrain some
+behavior that can be seen as a false positive for some users. Adding a configuration is done 
+in the following steps:
+
+1. Adding a new configuration entry to [clippy_lints::utils::conf](/clippy_lints/src/utils/conf.rs)
+    like this:
+    ```rust
+    /// Lint: LINE_NAME. <The configuration field doc comment>
```
Comment (llogiq) [inline doc/adding_lints.md:540]: What does `LINE_NAME` signify here?

Comment (xFrednet) [reply doc/adding_lints.md:540]: The `LINT_NAME` stands for the lint that the configuration is linked to. The `LINT_NAME` identifier is also later used for the lint. Do you think I should clarify it? :upside_down_face:

Comment (llogiq) [reply doc/adding_lints.md:540]: Ah, then you should probably just rename it from `LINE_NAME` to `LINT_NAME`.

Comment (xFrednet) [reply doc/adding_lints.md:540]: Oh, well reading is not always my strong suite :sweat_smile: I'll update it right away :upside_down_face:

**Comment by llogiq** [comment]
Thank you! @bors r+

File: doc/basics.md [hunk 2]
```diff
@@ -94,3 +95,22 @@ cargo dev ra_setup
...
+| UB           | Undefined Behavior                     |
+| FP           | False Positive                         |
+| FN           | False Negative                         |
+| ICE          | Internal Compiler Error                |
+| AST          | Abstract Syntax Tree                   |
+| MIR          | Mid-Level Intermediate Representation  |
+| HIR          | High-Level Intermediate Representation |
+| TCX          | Type context                           |
+
+This is a concise list of abbreviations that can come up during clippy development. An extensive
```
Comment (flip1995) [inline doc/basics.md:112]:
```suggestion
This is a concise list of abbreviations that can come up during Clippy development. An extensive
```

File: doc/adding_lints.md [hunk 3]
```diff
@@ -526,6 +527,81 @@ Before submitting your PR make sure you followed all of the basic requirements:
...
+        }
+        ```
+3. Passing the configuration value to the lint impl struct:
+
+    First find the struct construction in the [clippy_lints lib file](/clippy_lints/src/lib.rs). 
+    Make sure that `clippy dev update_lints` added it beforehand. The configuration value is now 
+    cloned or copied into a local value that is then passed to the impl struct like this:
+    ```rust
+    // Default generated registration:
+    store.register_late_pass(|| box module::StructName);
```
Comment (flip1995) [inline doc/adding_lints.md:588]: `update_lints` doesn't add `register_*_pass` to the `lib.rs` file, so this is in contradiction to earlier in the document.

Although, when adding a configuration value to a lint the lint registration is usually already there.

**Review by flip1995 — COMMENTED** [review]
Found 2 NITs

**Comment by xFrednet** [comment]
I seem to always mess a small thing up in these PRs. 😅 Oh well that's what reviews are for. Thank you both for reviewing these changes 🙃.

I'll correct the last two nits later today.

**Pull request merged**